primitives/test-helpers/src/lib.rs

```diff
@@ -0,0 +1,215 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+#![forbid(unused_crate_dependencies)]
+#![forbid(unused_extern_crates)]
+
+//! A set of primitive constructors, to aid in crafting meaningful testcase while reducing repetition.
+//!
+//! Note that `dummy_` prefixed values are meant to be fillers, that should not matter, and will
+//! contain randomness based data.
+use polkadot_primitives::v1::{
+	CandidateCommitments, CandidateDescriptor, CandidateReceipt, CollatorId, CollatorSignature,
+	CommittedCandidateReceipt, Hash, HeadData, Id as ParaId, ValidationCode, ValidationCodeHash,
+	ValidatorId,
+};
+use sp_application_crypto::sr25519;
+use sp_keyring::Sr25519Keyring;
+use sp_runtime::generic::Digest;
+
+/// Creates a candidate receipt without
+pub fn dummy_candidate_receipt<H: AsRef<[u8]>>(relay_parent: H) -> CandidateReceipt<H> {
+	CandidateReceipt::<H> {
+		commitments_hash: dummy_candidate_commitments(dummy_head_data()).hash(),
+		descriptor: dummy_candidate_descriptor(relay_parent),
+	}
+}
+
+/// Creates a committed candidate receipt without
+pub fn dummy_committed_candidate_receipt<H: AsRef<[u8]>>(
+	relay_parent: H,
+) -> CommittedCandidateReceipt<H> {
+	CommittedCandidateReceipt::<H> {
+		descriptor: dummy_candidate_descriptor::<H>(relay_parent),
+		commitments: dummy_candidate_commitments(dummy_head_data()),
+	}
+}
+
+pub fn dummy_candidate_receipt_bad_sig(
+	relay_parent: Hash,
+	commitments: impl Into<Option<Hash>>,
+) -> CandidateReceipt<Hash> {
+	let commitments_hash = if let Some(commitments) = commitments.into() {
+		commitments
+	} else {
+		dummy_candidate_commitments(dummy_head_data()).hash()
+	};
+	CandidateReceipt::<Hash> {
+		commitments_hash,
+		descriptor: dummy_candidate_descriptor_bad_sig(relay_parent),
+	}
+}
+
+pub fn dummy_candidate_commitments(head_data: impl Into<Option<HeadData>>) -> CandidateCommitments {
+	CandidateCommitments {
+		head_data: head_data.into().unwrap_or(HeadData(vec![])),
+		upward_messages: vec![],
+		new_validation_code: None,
+		horizontal_messages: vec![],
+		processed_downward_messages: 0,
+		hrmp_watermark: 0_u32,
+	}
+}
+
+/// Create meaningless dummy hash.
+pub fn dummy_hash() -> Hash {
+	// TODO make this PRNG based
```

Comment:
@drahnr are you ok with leaving this out for now and leaving for a follow up?

Reply:
It would make things a little more complicated because we would no longer be able to use this when we are asserting the output

Reply:
Please remove the TODOs, it's fine for right now, and I think we should introduce a separate fn with `_rand` suffix